cloudsim-plus/src/main/java/org/cloudbus/cloudsim/power/PowerMeter.java

```diff
@@ -0,0 +1,158 @@
+package org.cloudbus.cloudsim.power;
+
+import org.cloudbus.cloudsim.core.CloudSimEntity;
+import org.cloudbus.cloudsim.core.CloudSimTags;
+import org.cloudbus.cloudsim.core.Simulation;
+import org.cloudbus.cloudsim.core.events.SimEvent;
+import org.cloudbus.cloudsim.power.models.PowerModel;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.cloudbus.cloudsim.core.CloudSimTags.POWER_MEASUREMENT;
+
+/**
+ * Periodically measures the current power usage of one or more {@link PowerAware} entities,
+ * according to a defined {@link #getMeasurementInterval() interval}, storing the results.
+ *
+ * @see #getPowerMeasurements()
+ * @since CloudSim Plus 6.0.0
+ */
+public class PowerMeter extends CloudSimEntity {
+
+    private final Callable<List<? extends PowerAware<? extends PowerModel>>> powerAwareEntitiesFn;
+
+    private double measurementInterval;
+    /** @see #getStartTime() */
+    private double startTime;
+
+    private final List<PowerMeasurement> powerMeasurements = new LinkedList<>();
+
+    /**
+     * Initializes a {@link PowerMeter} to measure power consumption of a single {@link PowerAware} entity.
+     */
+    public PowerMeter(final Simulation simulation, final PowerAware<? extends PowerModel> powerAwareEntity) {
+        this(simulation, new ArrayList<>(Arrays.asList(powerAwareEntity)));
+    }
+
+    /**
+     * Initializes a {@link PowerMeter} to measure the combined power consumption of a list of {@link PowerAware} entities.
+     *
+     * <p>If you want to compute power consumption individually for each entity,
+     * check {@link #PowerMeter(Simulation, PowerAware)}.</p>
+     */
+    public PowerMeter(final Simulation simulation, final List<? extends PowerAware<? extends PowerModel>> powerAwareEntities) {
+        this(simulation, () -> powerAwareEntities);
+    }
+
+    /**
+     * Initializes a {@link PowerMeter} with a function that supplies a list of {@link PowerAware} entities
+     * to have their combined power consumption measured.
+     * This is useful if the list of entities varies during a simulation run.
+     *
+     * <p>If you want to compute power consumption individually for each entity,
+     * check {@link #PowerMeter(Simulation, PowerAware)}.</p>
+     */
+    public PowerMeter(final Simulation simulation, final Callable<List<? extends PowerAware<? extends PowerModel>>> powerAwareEntitiesFn) {
+        super(simulation);
+        this.powerAwareEntitiesFn = powerAwareEntitiesFn;
+        this.startTime = 0;
+    }
+
+    @Override
+    protected void startEntity() {
+        schedule(this, startTime, POWER_MEASUREMENT);
```

Comment:
I've update the docs and performed small refactorings.

About this start time, just a little thing.
After the entity starts, changing the `startTime` has no effect.
This way, we could make the attribute final, remove the setter
and pass it for the constructors.

What about that?

Reply:
What do you think of removing it altogether? I think it's an unnecessary feature to initialize a PowerMeter to start in e.g. 10 seconds. If we include it we should also include an `endTime`.

In my opinion, users can create and kill a PowerMeter the moment they want to. There is no need to include this rare use case in the API.

Reply:
It's ok for me to remove it.

Reply:
I'll do that.